### ISSUE TITLE
Detecting kill signal by server when running on HPC

### ISSUE THREAD

**Issue by erfanshekarriz** [snakemake/snakemake#2633]
Hello! 

I love snakemake and have been using it to build many projects. 

One feature I wanted to request is the ability of snakemake to detect when a job has been killed by the server when running a command on the HPC. 

When the software I use exceeds the memory limit set by the resources ```mem_mb``` directive, our server automatically kills that job, but this signal is never received by snakemake. 

This can be frustrating because the ```attempt``` function when setting the memory, e.g.:
```mem_mb = lambda wildcards, attempt, threads: attempt * threads * 1 * 10**3```
does not work despite setting ```--retries 3``` when running snakemake, and the job keeps hanging. 

I then have to manually modify my snakefile to add memory each time but need to tune it back because this large amount of memory is only used up by certain and not all datasets.

Is there a way to automatically detect when a server has canceled a task? That would be very helpful. 

Best, 

Erfan

**Comment by kdm9** [comment]
This is the job of [`--cluster-generic-status-command`](https://snakemake.github.io/snakemake-plugin-catalog/plugins/executor/cluster-generic.html) and its equivalents. You should write a script that queries the cluster for a job's status, and snakemake will call that script to get the exit status etc for each cluster job (ideally use or adapt one from an [existing profile](https://github.com/snakemake-profiles/doc)).

**Comment by chrarnold** [comment]
Or use the --slurm directive directly, which detects such cases I believe in case you have a SLURM system.

**Comment by cmeesters** [comment]
Right, on an HPC cluster with SLURM and snakemake >=v8 (better most recent snakemake version and plugins), a cancelled job will be recognised as such.
